staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -389,10 +391,10 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 	} else {
 		// This never happens, because dedupDeltas never returns an empty list
 		// when given a non-empty list (as it is here).
-		// But if somehow it ever does return an empty list, then
-		// We need to remove this from our map (extra items in the queue are
-		// ignored if they are not in the map).
-		delete(f.items, id)
+		// But if somehow it ever did return an empty list, then
+		// it would not be legitimate to simply remove the entry for `id`
+		// from `f.items` without also ensuring that `id` is absent from `f.queue`.
```

Comment:
I think it's sufficient to say `This never happens, because ...` and then add a `panic()`. At least that would be more intuitive for me as a reader. 

What do you think?

Reply:
I think you are suggesting removing these lines:
```
		// But if somehow it ever did return an empty list, then
		// it would not be legitimate to simply remove the entry for `id`
		// from `f.items` without also ensuring that `id` is absent from `f.queue`.
```